### PR TITLE
allow for explicit initialization of SALT and SALTP

### PR DESCRIPTION
**Pull request by plgbrts**
This PR depends on https://github.com/OPM/opm-common/pull/2931

### CODE REVIEW

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -2571,6 +2573,12 @@ private:
         if (FluidSystem::enableVaporizedOil() && !has_rv)
             throw std::runtime_error("The ECL input file requires the RV keyword to be present if"
                                      " vaporized oil is enabled");
+        if (enableBrine && !has_salt)
+            throw std::runtime_error("The ECL input file requires the SALT keyword to be present if"
+                                     " brine is enabled");
```
Comment (blattms) [inline ebos/eclproblem.hh:2578]: Isn't there a change of semantics here?

It seems that now an error is thrown if you simulate brine and did not explicitly specify SALT (`has_salt==false`). I assume that previously the default would have been used.

**Review by blattms — COMMENTED** [review]
As I am far from being an expert for eclipse file, I might be wrong and the change is the right thing, but

File: ebos/eclproblem.hh (hunk 1 above)
Comment (plgbrts) [reply ebos/eclproblem.hh]: The SALT keyword is only required if brine is enabled _and_ the model is initialized  explicitly. The modified part of the code deals with the explicit initialization.  I have improved the error message.

**Comment by totto82** [comment]
jenkins build this please

**Comment by blattms** [comment]
Thanks @goncalvesmachadoc. Will do the update_data drill (quite uncommon for me).

**Comment by blattms** [comment]
jenkins build this update_data please

**Comment by blattms** [comment]
jenkins build this with opm-tests=3774

**Comment by akva2** [comment]
yes look at your trigger...

**Comment by akva2** [comment]
jenkins build this opm-tests=3774 please

**Comment by akva2** [comment]
argh, and i copied your second screw-up. let's try again.

**Comment by akva2** [comment]
jenkins build this opm-tests=727 please

**Comment by akva2** [comment]
yes, missing please. we parse between jenkins *uild this and please. and you referenced the opm-simulators pr number, not the opm-tests one. the with would have been harmless.

**Comment by blattms** [comment]
:flushed: Oh my...

**Pull request merged**